examples/cmake/FindGeodeNative.cmake

```diff
@@ -61,52 +60,83 @@ set(_GEODE_NATIVE_PATHS)
 
 if(_GEODE_NATIVE_ROOT)
     set(_GEODE_NATIVE_HINTS ${_GEODE_NATIVE_ROOT})
+    set(_GEODE_NATIVE_HINTS ${_GEODE_NATIVE_ROOT})
 else()
     set(_GEODE_NATIVE_PATHS (
         "/opt/local"
         "/usr/local"
-        "${CMAKE_CURRENT_SOURCE_DIR}/../../../../"
+        "${CMAKE_CURRENT_SOURCE_DIR}/../../../"
         "C:/program files" ))
 endif()
 
-set(_GEODE_NATIVE_NAMES apache-geode)
+# Begin - component "cpp"
+set(_GEODE_NATIVE_CPP_NAMES apache-geode)
 
-find_library(GEODE_NATIVE_LIBRARY
-    NAMES ${_GEODE_NATIVE_NAMES}
+find_library(GEODE_NATIVE_CPP_LIBRARY
+    NAMES ${_GEODE_NATIVE_CPP_NAMES}
     HINTS ${_GEODE_NATIVE_HINTS}
     PATHS ${_GEODE_NATIVE_PATHS}
-    PATH_SUFFIXES geode-native/lib
+    PATH_SUFFIXES geode-native/lib lib 
 )
 
-
-
 # Look for the header file.
-find_path(GEODE_NATIVE_INCLUDE_DIR NAMES geode/CacheFactory.hpp
+find_path(GEODE_NATIVE_CPP_INCLUDE_DIR NAMES geode/CacheFactory.hpp
     HINTS ${_GEODE_NATIVE_HINTS}
     PATHS ${_GEODE_NATIVE_PATHS}
-    PATH_SUFFIXES geode-native/include
+    PATH_SUFFIXES geode-native/include include
+)
+# End - component "cpp"
+
+# Begin - component "dotnet"
+set(_GEODE_NATIVE_DOTNET_NAMES Apache.Geode.dll)
+
+find_file(GEODE_NATIVE_DOTNET_LIBRARY
+  NAMES ${_GEODE_NATIVE_DOTNET_NAMES}
+  HINTS ${_GEODE_NATIVE_HINTS}
+  PATHS ${_GEODE_NATIVE_PATHS}
+  PATH_SUFFIXES geode-native/bin bin 
 )
+# End - component "dotnet"
 
 # TODO find version
 set(GEODE_NATIVE_VERSION_STRING 1.0)
 
+if (GeodeNative_FIND_COMPONENTS)
+  set(_GEODE_NATIVE_REQUIRED_VARS)
+  foreach (component ${GeodeNative_FIND_COMPONENTS})
+    if (component STREQUAL "cpp")
+      message(STATUS "component 'cpp'")
+      list(APPEND _GEODE_NATIVE_REQUIRED_VARS GEODE_NATIVE_CPP_LIBRARY GEODE_NATIVE_CPP_INCLUDE_DIR)
+    endif()
+    if (component STREQUAL "dotnet")
+      message(STATUS "component 'dotnet'")
+      list(APPEND _GEODE_NATIVE_REQUIRED_VARS GEODE_NATIVE_DOTNET_LIBRARY)
+    endif()
+  endforeach()
+endif()
+
 include(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(GEODE_NATIVE
-                                  REQUIRED_VARS GEODE_NATIVE_LIBRARY GEODE_NATIVE_INCLUDE_DIR
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(GeodeNative
+                                  REQUIRED_VARS ${_GEODE_NATIVE_REQUIRED_VARS}
                                   VERSION_VAR GEODE_NATIVE_VERSION_STRING)
 
 # Copy the results to the output variables and target.
-if(GEODE_NATIVE_FOUND)
-  set(GEODE_NATIVE_LIBRARIES ${GEODE_NATIVE_LIBRARY})
-  set(GEODE_NATIVE_INCLUDE_DIRS ${GEODE_NATIVE_INCLUDE_DIR})
-  set(GEODE_NATIVE_CPP_TARGET "apache.geode::cpp")
+if(GeodeNative_FOUND)
   if(NOT TARGET ${GEODE_NATIVE_CPP_TARGET})
+    set(GEODE_NATIVE_CPP_TARGET "apache.geode::cpp")
```

Comment:
I think we should keep consistent naming. Maybe we should change this from `apache.geode::cpp` to `ApacheGeode::cpp` to match `find_package(ApacheGeode COMPONENT cpp)`.